src/peer_connection.cpp

```diff
@@ -355,6 +355,18 @@ namespace libtorrent {
 					peer_log(peer_log_alert::outgoing, "SET_TOS", "tos: %d e: %s"
 						, m_settings.get_int(settings_pack::peer_tos), ec.message().c_str());
 				}
+#endif
+#ifdef IP_DSCP_TRAFFIC_TYPE
```

Comment:
This wants to be restructured to

```
    if (m_settings.get_int(settings_pack::peer_tos) != 0)
    {
        if (is_v4())
            type_of_service(val)
#ifdef IPV6_TCLASS
        else if (is_v6())
            traffic_class(val)
#endif
#ifdef IP_DSCP_TRAFFIC_TYPE
        dscp_traffic_type(val >> 2)
#endif
    }
```

Or maybe an if/else on the `IP_DSCP_TRAFFIC_TYPE`? Only attempt one form?